packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -317,13 +317,16 @@ class Auth {
         const refreshToken = refreshTokenRegistered || refreshTokenGuest
         if (refreshToken) {
             try {
-                return this.queueRequest(
+                return await this.queueRequest(
                     () => helpers.refreshAccessToken(this.client, {refreshToken}),
                     !!refreshTokenGuest
                 )
-            } catch {
-                // If anything bad happens during refresh token flow
-                // we continue with the PKCE guest user flow.
+            } catch (error) {
+                // If the refresh token is invalid, we need to re-login the user
+                if (error instanceof Error && error.message === '400 Bad Request') {
+                    // clean up storage and restart the login flow
+                    await this.logout()
```

Comment:
I think we need to return here because `logout` calls `loginGuestUser`, if we don't return, `loginGuestUser` will be called again.

```suggestion
                    return this.logout()
```

Reply:
So, there is a TODO for that logout call, and we expect it to call the /logout endpoint.

```
        // TODO: are we missing a call to /logout?
        // Ticket: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001EFF4nYAH/view
```

I'm curious what happens when you call `/logout` with an expired access token? 🤔

Reply:
I don't think we need to, the logout has returned the loginGuestUser in there